### DISCUSSION TITLE
SD3.5 Support

### DISCUSSION THREAD

**Discussion by Acly** [Acly/krita-ai-diffusion#1328] [Announcements]
## Stable Diffusion 3.5: Current State

Plugin version: [1.27.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.27.0)

### _Experimental Support_

It's new, so there aren't any inpaint, control or other supplementary models out yet to support all the features. But you can try out the base model and LoRA already.

## Model Download

Currently SD3.5 Large (8B) is released. It typically comes as a checkpoint with integrated VAE, but no text encoders. That means text encoders have to be downloaded separately.

### Checkpoint

 There is a smaller fp8 version which required less VRAM:
* [sd3.5_large_fp8_scaled.safetensors](https://huggingface.co/Comfy-Org/stable-diffusion-3.5-fp8/resolve/main/sd3.5_large_fp8_scaled.safetensors)
* _or_ use the official files at [huggingface/stabilityai](https://huggingface.co/stabilityai/stable-diffusion-3.5-large)

The models must be placed into the `ComfyUI/models/checkpoints` folder. 

### Text Encoders
SD3.5 supports up to 3 text encoders.

* [Clip-L (clip_l.safetensors)](https://huggingface.co/comfyanonymous/flux_text_encoders/resolve/main/clip_l.safetensors) 
* [Clip-G (clip_g.safetensors)](https://huggingface.co/Comfy-Org/stable-diffusion-3.5-fp8/resolve/main/text_encoders/clip_g.safetensors) 
* [T5 XXL (t5xxl_fp8_e4m3fn.safetensors)](https://huggingface.co/comfyanonymous/flux_text_encoders/resolve/main/t5xxl_fp8_e4m3fn.safetensors) 

Text encoder models must be placed into the `ComfyUI/models/clip` folder.

Note that T5 is _optional_. If you don't have it, the model will still work, but may have slightly reduced text understanding.

If you already have these models (eg. because you are using Flux), you don't have to download them again. They can be shared. GGUF variants of T5 also work.

## Future Work
I didn't get around to do much testing yet. Let me know what you find.

**Comment by XoseNomad** [comment]
Is it possible to add support for SD 3.5 medium? I was trying to use it but it won't work. It is significantly quicker than large model.

**Reply by Neeqstock** [reply]
Same. 
I have all the clip files in the correct folders.
I get this error with SD 3.5 medium. Didn't try with large because my GPU probably won't be able to support it.
(P.S.: it would be awesome if GGUF versions of checkpoints could be supported :))

![immagine](https://github.com/user-attachments/assets/2e016388-bea3-45a5-abd7-fc9b4d19101b)

**Comment by Ckwaca** [comment]
I install the sd3.5_large-Q8_0.gguf vers (same result than original large) from city 96 (on a  rtx 4060 8gv ram it work well)
https://huggingface.co/city96/stable-diffusion-3.5-large-gguf/tree/main

I just had to download the VAE "diffusion_pytorch_model.safetensors" rename it sd3.safetensors
https://huggingface.co/stabilityai/stable-diffusion-3.5-large/tree/main/vae

I add the 3 SD3.5 supports up to 3 text encoders. in first post

and it work well. I don't know if i will be able to do everything on krita like SDXL, but the image generation is good take 2 time less than Flux guff 4 KS
I set it in euler
![image](https://github.com/user-attachments/assets/cdf1ccd5-1499-4f50-9835-5d613878131c)

Exemple  test promp i find
Photo of a woman with pink hair raising her left hand above her head. Stand with one leg on a hardwood floor.

Flux
![flux](https://github.com/user-attachments/assets/9a438efa-5c9c-4f27-8629-34354886fceb)

SD 3.5
![test 3 5](https://github.com/user-attachments/assets/22d4775d-a49c-4355-b079-a0ec122d28df)

SDXL get issue with face and pose not "natural" , i find the test prompt i used (https://stable-diffusion-art.com/sdxl-vs-flux/) and SDXL get issue with palm hand, SD 3.5 no issue 
![image](https://github.com/user-attachments/assets/ba5df440-abf9-4210-b382-538ade380864)

I will see if i can install SDXL linghting, cos look like much better than default model in krita ai diffusion even if they are already nice

**Comment by guiweber** [comment]
So I've tried to get SD 3.5 running but I'm getting strange distorted images no matter the prompt... here is one example for "woman with blue hair lying in the grass smiling"

![image](https://github.com/user-attachments/assets/4c159c2c-e8a3-4391-8370-41405de177e7)


I have all three text encoders installed and I've tried both sd3.5_large_fp8_scaled.safetensors and the official sd3.5_large.safetensors but they give similarly distorted results...

Any idea of what is wrong?

I'm running SDXL and Flux without any issues.

**Reply by eightball-unlock** [reply]
Change your Sampler to 'SD3 - DPM++ 2M Uniform'.

**Reply by guiweber** [reply]
Thanks!

**Comment by Streamtabulous** [comment]
with 1.38 fresh install sd3 stopped working any ideas how to get it back?
![Screenshot 2025-08-27 193518](https://github.com/user-attachments/assets/921d324f-0711-41c4-ba12-a4fd6eb30461)
